### ISSUE TITLE
Strange behavior of widget.form

### ISSUE THREAD

**Issue by tandem97** [fyne-io/fyne#2466]
### Describe the bug:
I have a form with 2 labels and 2 circles. First of all i cant understand why the position of this circles is so strange. They just lay on the top of a form cell. Second problem is when i change theme form text just disappeared. And last, the cirle form depends on the window size on application start, so cicrle can be not circle)

### To Reproduce:
just go run this code

### Screenshots:
![Безымянный](https://user-images.githubusercontent.com/82098282/133132207-b030c60a-aee8-4838-b327-1cad0304b5b4.png)
![Снимок экрана 2021-09-13 205033](https://user-images.githubusercontent.com/82098282/133132338-b9ff5e58-5401-4a2c-ba8b-1e6f27e0c40f.png)

### Example code:
```go
package main

import (
	"image/color"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/canvas"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/layout"
	"fyne.io/fyne/v2/theme"
	"fyne.io/fyne/v2/widget"
)

func main() {

	app := app.New()

	mainWindow := app.NewWindow("Hello")
	mainWindow.CenterOnScreen()
	mainWindow.Resize(fyne.NewSize(50, 50))
	mainWindow.SetIcon(theme.FyneLogo())

	darkThemeMenuItem := fyne.NewMenuItem("Dark", func() { app.Settings().SetTheme(theme.DarkTheme()) })
	lightThemeMenuItem := fyne.NewMenuItem("Light", func() { app.Settings().SetTheme(theme.LightTheme()) })
	themeMenuItem := fyne.NewMenuItem("Theme", nil)
	themeMenuItem.ChildMenu = fyne.NewMenu("Theme", darkThemeMenuItem, lightThemeMenuItem)
	settingsMenu := fyne.NewMenu("Setting", themeMenuItem)
	mainMenu := fyne.NewMainMenu(settingsMenu)
	mainWindow.SetMainMenu(mainMenu)

	form := &widget.Form{Items: []*widget.FormItem{{Text: "test1", Widget: canvas.NewCircle(color.Black)}, {Text: "test2", Widget: canvas.NewCircle(color.Black)}}}
	buttonTest := widget.NewButton("test", func() { form.Refresh() })

	mainLayout := container.New(layout.NewBorderLayout(nil, nil, nil, buttonTest), buttonTest, form)

	mainWindow.SetContent(mainLayout)

	mainWindow.ShowAndRun()
}

```

### Device (please complete the following information):
 - win10
 - go 1.17
 - fyne 2.0.4

**Comment by andydotxyz** [comment]
The circle primitive is perhaps badly named, but it (like all others) will fill space allocated, so it will stretch to non-circle if you ask it to be wider than tall - which is what the form content is doing.

It seems that the form widget does not have a correct layout when the items on the right are not widgets.
This is a bug.
Plus not updating on theme change is a bug, well done for finding that.

**Comment by tandem97** [comment]
> The circle primitive is perhaps badly named, but it (like all others) will fill space allocated, so it will stretch to non-circle if you ask it to be wider than tall - which is what the form content is doing.
> 
> It seems that the form widget does not have a correct layout when the items on the right are not widgets.
> This is a bug.
> Plus not updating on theme change is a bug, well done for finding that.

![image](https://user-images.githubusercontent.com/82098282/133141652-728750b7-a76e-4fdf-af0e-dae57ebce4ab.png)
this is the same circle but with another initial size ```mainWindow.Resize(fyne.NewSize(1200, 1200))```
![image](https://user-images.githubusercontent.com/82098282/133141929-8692b5c9-7688-454a-b13c-7a3492758bd7.png)
this circle with window initial size 50 50 ```mainWindow.Resize(fyne.NewSize(50, 50))```

They are different, but i stretch form to the same size

**Comment by andydotxyz** [comment]
Oh I see, that sounds like a bug too, but rendering was updated recently so perhaps that item is resolved in `v2.1.0-rc1`

**Comment by tandem97** [comment]
> Oh I see, that sounds like a bug too, but rendering was updated recently so perhaps that item is resolved in `v2.1.0-rc1`

Yes in 2,1 it is ok!

**Comment by andydotxyz** [comment]
> want 2,1 stable

Then please help us with testing :) on gophers Slack server we run the vote - it will run until Thursday to check there are no blocking issues.

**Comment by andydotxyz** [comment]
This should have been resolved in the 2.1 release - please update to `v2.1.0` version and re-open if the issue remains.
